### PR TITLE
chore(ci): request necessary permissions

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #1172 
Related to: [#1783](https://github.com/cryostatio/cryostat/issues/1783)

## Description of the change:
*This change adds allows a match expression example to be copied to the clipboard...*

## Motivation for the change:
*This change is helpful because users may want to copy the example for easier use...*

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...*
2. *...*

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -76,6 +84,10 @@ jobs:
     strategy:
       matrix:
         node-version: [16.x, 18.x]
+    permissions:
+      statuses: write
+      pull-requests: write
+      packages: read
```
Comment (mwangggg) [inline .github/workflows/ci.yaml:90]: these should all be unnecessary? I don't think we're adding anything to the status checks or commenting on the PR in this job? Also the base permissions of the org should have `packages: read` by default

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -62,6 +66,10 @@ jobs:
...
```
Comment (mwangggg) [inline .github/workflows/ci.yaml:72]: same here as above

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -34,6 +36,8 @@ jobs:
     strategy:
       matrix:
         node-version: [16.x, 18.x]
+    permissions:
+      pull-requests: read
```
Comment (mwangggg) [inline .github/workflows/ci.yaml:40]: is this necessary? I don't see any step that requires this permission

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -20,6 +20,8 @@ jobs:
...
```
Comment (mwangggg) [inline .github/workflows/ci.yaml:24]: same here as above.. unless I'm missing something?

File: .github/workflows/dependent-issues.yml [hunk 5]
```diff
@@ -20,6 +20,10 @@ on:
 
 jobs:
   check:
+    permissions:
+      issues: write
+      pull-requests: write
+      statuses: write
```
Comment (mwangggg) [inline .github/workflows/dependent-issues.yml:26]: I believe these are all unnecessary too? should be fine with just the default permissions we have

Comment (aali309) [reply .github/workflows/dependent-issues.yml:26]: [https://github.com/cryostatio/cryostat-web/actions/runs/6952595800/job/18916412442?pr=1153](https://github.com/cryostatio/cryostat-web/actions/runs/6952595800/job/18916412442?pr=1153)
I thought this was a permission issue?

**Comment by mwangggg** [comment]
yes yes that was me being silly

File: .github/workflows/semantic-pr.yml [hunk 6]
```diff
@@ -11,6 +11,9 @@ on:
 jobs:
   main:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      statuses: write
```
Comment (mwangggg) [inline .github/workflows/semantic-pr.yml:16]: no need for statuses

File: .github/workflows/dependent-issues.yml (hunk 5 above)
Comment (mwangggg) [reply .github/workflows/dependent-issues.yml:26]: I don't think we need statuses here...

Comment (mwangggg) [reply .github/workflows/dependent-issues.yml:26]: nvm I'm silly

**Review by mwangggg — APPROVED** [review]
LGTM :)

**Review by andrewazores — APPROVED** [review]

**Pull request merged**